### PR TITLE
[Cleanup]: Remove RECORD_ARRAY_MANAGER_IDENTIFIERS

### PR DESCRIPTION
**Pull request by snewcomer**
This flag helped us move from using Internal Models to Identifiers.  With the 3.23 release, we can remove this flag.

### CODE REVIEW

**Comment by snewcomer** [comment]
A few possible paths here.  1 - we could consider this feature flag done and any further work to remove internalModel could be under new feature flags.  2 - close this PR and find more spots to remove internalModel (core store?).

One reason we might consider this feature flag completed is because we have ✂️  the use of `internalModel` in record-array and associated modules.

File: packages/canary-features/addon/default-features.ts [hunk 1]
```diff
@@ -177,6 +177,5 @@ export default {
   REQUEST_SERVICE: null,
   CUSTOM_MODEL_CLASS: null,
   FULL_LINKS_ON_RELATIONSHIPS: true,
-  RECORD_ARRAY_MANAGER_IDENTIFIERS: true,
   REMOVE_RECORD_ARRAY_MANAGER_LEGACY_COMPAT: null,
```
Comment (igorT) [inline packages/canary-features/addon/default-features.ts:180]: We probably want to keep the flag exposed

File: packages/canary-features/addon/index.ts [hunk 2]
```diff
@@ -38,7 +38,6 @@ export const REQUEST_SERVICE = featureValue(FEATURES.REQUEST_SERVICE);
 export const IDENTIFIERS = featureValue(FEATURES.IDENTIFIERS);
 export const CUSTOM_MODEL_CLASS = featureValue(FEATURES.CUSTOM_MODEL_CLASS);
 export const FULL_LINKS_ON_RELATIONSHIPS = featureValue(FEATURES.FULL_LINKS_ON_RELATIONSHIPS);
```
Comment (igorT) [inline packages/canary-features/addon/index.ts:40]: We probably want to keep the flag exposed

File: packages/store/addon/-private/system/references/reference.ts [hunk 3]
```diff
@@ -32,15 +32,10 @@ function isResourceIdentiferWithRelatedLinks(
   return value && value.links && value.links.related;
 }
 
-// TODO: simplify after 3.23 release and only store identifier
-export const REFERENCE_CACHE = new WeakMap<Reference, InternalModel | StableRecordIdentifier>();
+export const REFERENCE_CACHE = new WeakMap<Reference, StableRecordIdentifier>();
```
Comment (igorT) [inline packages/store/addon/-private/system/references/reference.ts:35]: Thanks for the cleanup

**Comment by igorT** [comment]
@snewcomer we should use a new feature flag for new units of work, so they are easier to back out of. A lot of cleanup might not even need it, depending on how risky it is

**Pull request merged**